examples/tests/valid/inflight_closure_as_super_param.test.w

```diff
@@ -0,0 +1,35 @@
+class Foo {}
+
+class Base {
+  pub h: inflight (): str;
+  pub f_base: Foo;
+  new(handler: inflight (): str, f: Foo) { 
+    this.h = handler;
+    this.f_base = f;
+  }
+}
+
+class Derived extends Base {
+  pub f: Foo;
+  new() {
+    super(inflight (): str => { 
+      return "boom!";
+    }, new Foo() as "in_root");
```

Comment:
For the case of arbitrary preflight classes, the issue of not being able to pass a newly-constructed class to super() is interesting because the normal construction pattern (where the child class receives the "this" scope) also isn't possible in TypeScript:

```ts
class MyConstruct extends Construct {
  constructor(scope: Construct, id: string, helper: Construct) {
    super(scope, id);
    // do something with helper...
  }
}

class MySuperConstruct extends MyConstruct {
  constructor(scope: Construct, id: string) {
    let c = new Construct(this, "Helper"); // error: 'super' must be called before accessing this
    super(scope, id, c);
  }
}
```

As you observed, it's possible to reuse the "scope" to make it work:

```ts
let c = new Construct(scope, "Helper");
```

But it's worth noting this technically changes the tree structure so the construct being created is now a sibling instead of a child. Not necessarily a big deal.

Having the compiler do this automatically sounds like a fair compromise given that I don't think there's any other workaround. Indeed, I think the only other way to access the implicit "$scope" argument in Winglang right now requires the `nodeof()` function right now - but `nodeof` expects a scope, so it doesn't really help:

```
class A {
  new(bucket: cloud.Bucket) {
  }
}

class B extends A {
  new() {
    super(new cloud.Bucket() in nodeof(this).scope!); // error: using this before "super" has been called
  }
}
```

I feel like it could be worth documenting in the language reference or [here](https://github.com/winglang/wing/blob/main/docs/docs/02-concepts/02-application-tree.md) that classes constructed inside `super()` statements will be siblings instead of children in the tree - what do you think?

Reply:
:+1:  Agree. I see the docs only contain information about preflight class "naming" and not about "scoping", I'll add a section and including this edge case.